src/main/webapp/script.js

```diff
@@ -0,0 +1,162 @@
+const MONTHS = ['January', 'February', 'March', 'April', 'May', 'June', 'July',
+                'August', 'September', 'October', 'November', 'December'];
+const OCD_STATE_TITLE = 'state:';
+const OCD_DISTRICT_TITLE = 'district:';
+const STATE_CODE_LENGTH = 2;
+const ELECTION_LIST_TITLE = 'Elections coming up in ';
+const COUNTRY_NAME = ' the United States';
+const NO_ELECTIONS_TEXT = 'No elections coming up soon.';
+const LEARN_MORE_BUTTON_TEXT = 'Learn more';
+const INVALID_DATE_TEXT = 'Invalid date';
+
+/**
+ * Fetch the list of upcoming elections and display relevant ones according
+ * to the state selected by the user.
+ */
+function listElections() {
+  let dropdownSelection = document.getElementById('select-state');
+  let selectedStateId = dropdownSelection.options[dropdownSelection.selectedIndex].value;
+  let selectedStateName = dropdownSelection.options[dropdownSelection.selectedIndex].text;
+  let nationalElections = [];
+  let stateElections = [];
+
+  if (selectedStateId === undefined) {
+    return;
+  }
+
+  fetch('/election')
+    .then(response => response.json())
+    .then((textResponse) => {
+      let electionList = JSON.parse(textResponse).elections;
+
+      // For every election returned, store it if the state matches or if national election,
+      // to later display the details to the user.
+      electionList.forEach((election) => {
+        let ocdId = election.ocdDivisionId;
+        let stateId = ocdId.indexOf(OCD_STATE_TITLE);
+        let districtId = ocdId.indexOf(OCD_DISTRICT_TITLE);
+
+        // ocdDivisionId is in the form "ocd-division/country:us/state:<state_code>"
+        // For example, Lousiana's ID is "ocd-division/country:us/state:la"
+        if (stateId !== -1) {
+          // If the state text is found in ocdDivisionId, then this is a state election.
+          if (ocdId.substring(stateId, stateId + OCD_STATE_TITLE.length + STATE_CODE_LENGTH)
```

Comment:
No need to change this, but normally we would use regular expressions (regex) for this, especially since the format is fairly straight forward.
The pros of regex is you can actually extract the string you are looking for directly, if your search expression matches.   Unfortunately for more complicated patterns, regex can get pretty tricky and difficult to get right.

Some more info here: https://www.regular-expressions.info/

Happy to chat more if you're interested in using regex.